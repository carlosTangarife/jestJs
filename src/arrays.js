export const arrayFruits = () => ['Mango', 'manzana', 'naranja', 'limon', 'pepino', 'banana'];

export const arrayColors = () => ['red', 'blue', 'yellow', 'green', 'orange'];