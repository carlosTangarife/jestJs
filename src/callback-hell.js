export function callBackHell(callback) {
    callback("Hola Javascripters");
}