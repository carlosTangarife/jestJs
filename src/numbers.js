export const numbers = (num1, num2) => {
    return num1 + num2;
}