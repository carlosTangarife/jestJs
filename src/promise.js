import axios from 'axios';

export const getDatafromApi = (url) => {
    return axios.get(url);
}