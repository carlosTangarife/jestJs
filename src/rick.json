{
    "id": 1,
    "gender": "Male",
    "name": "Ricky",
    "species": "Human",
    "status": "Alive"
}