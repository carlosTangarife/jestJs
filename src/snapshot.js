export const getCharacter = (data) => {
    const { gender, id, name, species, status } = data;
    return {
        id,
        gender,
        name,
        species,
        status
    }
}