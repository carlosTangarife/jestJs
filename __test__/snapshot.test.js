import { getCharacter } from '../src/snapshot';
import rick from '../src/rick.json';

describe('Trabajando con snapshot', ()=> {
    test('snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    })

    test('snapshot con excepciones', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            date: new Date(),
            name: "Carlos",
            lastname: "Tangarife"
        };

        expect(user).toMatchSnapshot();
    })

    test('snapshot sin excepciones', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            date: new Date(),
            name: "Carlos",
            lastname: "Tangarife"
        };

        expect(user).toMatchSnapshot({
            id: expect.any(Number),
            date: expect.any(Date)
        });
    })
})
