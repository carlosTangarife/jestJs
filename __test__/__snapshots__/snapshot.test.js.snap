// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trabajando con snapshot snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Ricky",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Trabajando con snapshot snapshot con excepciones 1`] = `
Object {
  "date": 2020-05-28T02:18:30.242Z,
  "id": 18,
  "lastname": "Tangarife",
  "name": "Carlos",
}
`;

exports[`Trabajando con snapshot snapshot sin excepciones 1`] = `
Object {
  "date": Any<Date>,
  "id": Any<Number>,
  "lastname": "Tangarife",
  "name": "Carlos",
}
`;
